hnn_core/parallel_backends.py

```diff
@@ -65,6 +65,9 @@ def _gather_trial_data(sim_data, net, n_trials, postproc):
         net.cell_response.update_types(net.gid_ranges)
         net.cell_response._vsoma.append(spikedata[3])
         net.cell_response._isoma.append(spikedata[4])
+        assert len(net.lfp) == len(spikedata[5])
```

Comment:
can this assert be removed?

Reply:
I suppose that's better suited in the testing code. The main motivation for this check is to make sure recordings aggregated during MPI runs are formatted correctly.

Reply:
yeah, assert statements in user code means a bug has gone without fixing. if an exception can be identified, it should be handled properly so as to not crash the code. they are useful in tests because they tell developers to fix the code (the user isn't expected to go into the source code).

Reply:
Absolutely this should go into the test